### PR TITLE
Refactor FullMeshLinked and ConnectAll()

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

File: p2p/net/mock/mock_test.go [hunk 1]
```diff
@@ -273,10 +273,12 @@ func TestNetworkSetup(t *testing.T) {
 
 func TestStreams(t *testing.T) {
 
-	mn, err := FullMeshConnected(context.Background(), 3)
+	mn, err := WithNPeers(context.Background(), 3)
```
Comment (jbenet) [inline p2p/net/mock/mock_test.go]: I dont see what we lose by continuing to use `FullMeshConnected` in cases like these. Maybe just make `FullMeshConnected` be:
```go
func FullMeshConnected(...) {
  mn, err := WithNPeers(context.Background(), 3)
  if err != nil {
    nil, err
  }
  mn.LinkAll()
  mn.ConnectAll()
  return mn, nil
}
```

 i think this is different from @whyrusleeping's case.

Comment (rht) [inline p2p/net/mock/mock_test.go]: Was thinking of doing this one instead.

**Comment by rht** [comment]
Looks like there are 2 different `ConnectAll()` depending on how `n1 == n2` is handled. Which one should be the default? The case where a node connects with itself (`continue`) is more common.

**Comment by jbenet** [comment]
@rht i think some tests are counting on avoiding duplicates. maybe pass in a bool? or just have two functions? `ConnectDistinctPairs()` ?

**Comment by rht** [comment]
`ConnectAllButSelf()`

**Comment by rht** [comment]
Since `ConnectAll()` is never used directly, it is not defined.

**Comment by jbenet** [comment]
@rht LGTM

**Pull request merged**